src/ensembl/src/shared/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import defaultStyles from './Checkbox.scss';
+
+type WithoutLabelProps = {
+  onChange: (status: boolean) => void;
+  classNames?: any;
+  disabled?: boolean;
+  checked: boolean;
+};
+
+type WithLabelProps = WithoutLabelProps & {
+  label: string;
+  labelClassName?: string;
+};
+
+type Props = WithLabelProps | WithoutLabelProps;
+
+const isWithLabel = (props: Props): props is WithLabelProps => {
+  return 'label' in props;
+};
+
+const Checkbox = (props: Props) => {
+  const [checkedStatus, toggleCheckedStatus] = useState(props.checked);
+
+  const handleOnChange = () => {
+    // Do nothing if it is disabled
+    if (props.disabled) {
+      return;
+    }
+
+    // Toggle the checked status
+    toggleCheckedStatus(!checkedStatus);
```

Comment:
If the `checked` prop is coming from the props, is there any point in keeping a parallel `checkedStatus` value in the state? Why not just rely on the `checked` value from the props?

Reply:
The `checked` parameter is optional so it may not always be passed from the parent component.